pkg/render/dex.go

```diff
@@ -138,11 +138,10 @@ func (c *dexComponent) Objects() ([]client.Object, []client.Object) {
 	// TODO the RequiredSecrets in the dex condig to not pass back secrets of this type.
 	if !c.cfg.DeleteDex {
```

Comment:
@rene-dekker DeleteDex is set to true only when (authentication.Spec.OIDC != nil && authentication.Spec.OIDC.Type == operatorv1.OIDCTypeTiger) [here](https://github.com/tigera/operator/blob/598cdf09604312d2916c917a5d7d6eef75c0c8a3/pkg/controller/utils/utils.go#L901), which is logically equivalent to !(authentication.Spec.OIDC == nil || authentication.Spec.OIDC.Type != operatorv1.OIDCTypeTiger). Therefore, it should be ok to included under the same if condition.